zenoh-flow/src/model/record/dataflow.rs

```diff
@@ -36,9 +36,9 @@ pub struct DataFlowRecord {
     pub operators: HashMap<NodeId, OperatorRecord>,
     pub sinks: HashMap<NodeId, SinkRecord>,
     pub sources: HashMap<NodeId, SourceRecord>,
-    pub connectors: Vec<ZFConnectorRecord>,
+    pub connectors: HashMap<NodeId, ZFConnectorRecord>,
     pub links: Vec<LinkRecord>,
-    pub counter: u32,
+    pub counter: usize,
```

Comment:
I believe `usize` is preferable when working with integers (as it will have the size of a machine word for the target architecture) which is why I went through with that modification. However, at least in this context, the performance gains are relatively limited (if there are any).

The limit should be implemented through some other means. I’ll revert the changes as the modification could be done the moment we decide to enforce a limit on the number of nodes / links.